modules/nf-core/vgan/haplocart/main.nf

```diff
@@ -0,0 +1,48 @@
+process VGAN_HAPLOCART {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda "bioconda::vgan=1.0.1"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/vgan:1.0.1--h9ee0642_0':
+        'quay.io/biocontainers/vgan:1.0.1--h9ee0642_0' }"
+
+    input:
+    tuple val(meta), path(reads)
+    path("hc_files")
+
+    output:
+    tuple val(meta), path("*[!posterior].txt") , emit: txt   
+    tuple val(meta), path("*.posterior.txt")    , emit: posterior
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def VERSION = '1.0.1'
+    def reads_args = meta.single_input_file ? "-fq1 ${reads}" : "-fq1 ${reads} -fq2 ${meta.reads2}"
+
+    """
+    vgan haplocart \\
+        $args \\
+        -t $task.cpus \\
+        $reads_args \\
+        -o ${prefix}.txt \\
+        --hc-files ${hc_files} \\
+        -pf ${prefix}.posterior.txt
+
+    echo $VERSION >> versions.yml
```

Comment:
This and the stub needs to follow the HEREDOC structure [^1]

[^1] https://nf-co.re/docs/contributing/modules#general ¶ 4